modules/aws-velero/iam.tf

```diff
@@ -54,3 +57,33 @@ resource "aws_iam_policy" "velero_backup" {
 }
 EOF
 }
+
+resource "aws_iam_role" "velero_backup" {
+  count = length(var.oidc_provider_url) != 0 ? 1 : 0
```

Comment:
Some brainstorming. Can we extract  length(var.oidc_provider_url) != 0 ? 1 : 0 to a  named local value?